src/libraries/Microsoft.PowerFx.Core/Entities/External/TableDelegationInfo.cs

```diff
@@ -0,0 +1,303 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.PowerFx.Core.Functions.Delegation;
+using Microsoft.PowerFx.Core.Utils;
+
+namespace Microsoft.PowerFx.Core.Entities
+{
+    // Supports delegation information for CDP connectors
+    public abstract class TableDelegationInfo
+    {
+        // Defines unsortable columns or columns only supporting ascending ordering
+        public SortRestrictions SortRestriction { get; init; }
+
+        // Defines columns that cannot be sorted and required properties
+        public FilterRestrictions FilterRestriction { get; init; }
+
+        // Used to indicate whether this table has selectable columns
+        public SelectionRestrictions SelectionRestriction { get; init; }
+
+        // Defines ungroupable columns
+        public GroupRestrictions GroupRestriction { get; init; }
+
+        // Filter functions supported by all columns of the table
+        // Those entries cumulate with the onee defined at column level
+        // Possible strings are defined in DelegationMetadataOperatorConstants
+        // Values not in this list are ignored
+        public IEnumerable<string> FilterFunctions { get; init; }
+
+        // Filter functions supported by the table
+        public IEnumerable<string> FilterSupportedFunctions { get; init; }
```

Comment:
Same here, enum instead of string. Also, how is this different from FilterFunctions?

Reply:
I have kept these original names and still trying to find someone able to explain those...
FilterFunctions is understood but this one is still mysterious
Am keeping it as it seems to play a role in delegation